### PR TITLE
feat(logging): add a sample for list logs

### PR DESCRIPTION
**Pull request by yucentao**
## Description

Fixes b/281691466

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: logging/list_logs_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	defer adminClient.Close()
+
+	// Create a log
+	logger := client.Logger(logID)
+	logger.Log(logging.Entry{Payload: "create a log"})
+	if err := logger.Flush(); err != nil {
+		log.Fatalf("logger.Flush() failed: %v", err)
+	}
+	// Delete the log
+	defer adminClient.DeleteLog(ctx, logID)
```
Comment (muncus) [inline logging/list_logs_test.go:63]: I dont think we need to clean up the log messages. The volume is pretty small, and removing this simplifies the test by eliminating `adminClient`.

File: logging/list_logs_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+func TestListLogs(t *testing.T) {
+	testutil.Retry(t, 6, 10*time.Second, func(r *testutil.R) {
+		buf := &bytes.Buffer{}
+		if err := listLogs(buf, projectID); err != nil {
+			r.Errorf("listLogs(%q) failed: %v", projectID, err)
+			return
+		}
+		if got, want := buf.String(), logID; !strings.Contains(got, want) {
```
Comment (muncus) [inline logging/list_logs_test.go:75]: This conditional is a little awkward to read.
Please either inline `buf.String()` and `logID` (the error message makes the order clear, so we dont need the specific variables `got` and `want`), or do the variable assignments before the conditional.

Comment (yucentao) [reply logging/list_logs_test.go:75]: This test pattern is suggested and widely used within Google codebase. go/go-test-examples#simple

**Comment by telpirion** [comment]
@yucentao Please do not provide links to internal Google resources in response to reviewer comments.

File: logging/list_logs_test.go (hunk 1 above)
Comment (yucentao) [reply logging/list_logs_test.go]: Done

File: logging/list_logs_test.go (hunk 2 above)
Comment (yucentao) [reply logging/list_logs_test.go]: Done

**Pull request merged**